src/title-buttons/titleButtonFromNarrow.js

```diff
@@ -3,51 +3,33 @@ import React from 'react';
 import type { ComponentType } from 'react';
 
 import type { Narrow } from '../types';
-import {
-  isHomeNarrow,
-  is1to1PmNarrow,
-  isGroupPmNarrow,
-  isSpecialNarrow,
-  isStreamNarrow,
-  isTopicNarrow,
-} from '../utils/narrow';
+import { caseNarrowDefault } from '../utils/narrow';
 import InfoNavButtonStream from './InfoNavButtonStream';
 import InfoNavButtonPrivate from './InfoNavButtonPrivate';
 import InfoNavButtonGroup from './InfoNavButtonGroup';
 import ExtraNavButtonStream from './ExtraNavButtonStream';
 import ExtraNavButtonTopic from './ExtraNavButtonTopic';
 
 type Props = $ReadOnly<{| color: string, narrow: Narrow |}>;
-type NarrowNavButton = ComponentType<Props>;
-type NarrowNavButtonCandidate = {|
-  isFunc: Narrow => boolean,
-  ButtonComponent: NarrowNavButton | null,
-|};
 
-const infoButtonHandlers: NarrowNavButtonCandidate[] = [
-  { isFunc: isHomeNarrow, ButtonComponent: null },
-  { isFunc: isSpecialNarrow, ButtonComponent: null },
-  { isFunc: isStreamNarrow, ButtonComponent: InfoNavButtonStream },
-  { isFunc: isTopicNarrow, ButtonComponent: InfoNavButtonStream },
-  { isFunc: is1to1PmNarrow, ButtonComponent: InfoNavButtonPrivate },
-  { isFunc: isGroupPmNarrow, ButtonComponent: InfoNavButtonGroup },
-];
+export const InfoButton: ComponentType<Props> = props =>
+  caseNarrowDefault(
+    props.narrow,
+    {
+      stream: streamName => <InfoNavButtonStream {...props} />,
+      topic: (streamName, topic) => <InfoNavButtonStream {...props} />,
+      pm: (emails, ids) =>
+        ids.length === 1 ? <InfoNavButtonPrivate {...props} /> : <InfoNavButtonGroup {...props} />,
+    },
+    () => false,
```

Comment:
`title buttons [nfc]: Use a more straightforward React idiom, and caseNarrow.`

This commit makes a nice simplification!

Returning `false` in the default of `caseNarrowDefault` is totally fine, and it's the most NFC-thing to do in this commit. But when we want to explicitly set the entire return value of a React component('s `render`) to be nothing, I think we often favor `null` over `false`. I don't feel very strongly either way (do you?), but this stood out to me just a tiny bit. 🙂

Reply:
I think it's fine. :slightly_smiling_face:

Certainly we often use `false` implicitly as a React node, in a form like `{someCondition() && <SomeComponent />}`. So I think the reader needs to be aware it's possible anyway.